### DISCUSSION TITLE
How to correctly pass arguments for useContractWrite()?

### DISCUSSION THREAD

**Discussion by NishantPacharne** [wevm/wagmi#878] [Q&A]
![image](https://user-images.githubusercontent.com/71060551/186403930-5d2acc32-034a-4d42-a98b-957c45a48822.png)

this is the mint function of my contract and I want to pass the quantity argument, please help me with that. my current code :- 
`
 const { writeAsync: mint, error: mintError } = useContractWrite({
    ...contractConfig,
    functionName: 'mint',
  });
`

and then Iam trying to add arguments this way :- 
`
const tx = await mint({args: [{_quantity: 1}]});
`

Thanks in advance!

**Comment by tmm** [comment]
`args` is an array of positional arguments. If you are using `0.8` and above, `args` type will be automatically inferred from your ABI.
